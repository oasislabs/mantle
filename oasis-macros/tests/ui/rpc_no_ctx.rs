#![feature(proc_macro_hygiene)]
#[oasis_std::service]
mod service {
    #[derive(Service, Default)]
    pub struct Counter(u32);

    impl Counter {
        pub fn new(ctx: &Context) -> Result<Self> {
            Ok(Default::default())
        }

        pub fn incr(&mut self, ctx: Context, amount: u32) -> Result<()> {
            self.0 += amount;
            Ok(())
        }
    }
}

fn main() {}
