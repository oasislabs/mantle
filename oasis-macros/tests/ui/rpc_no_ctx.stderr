error: Second argument to `Counter::incr` should be `&Context`.
  --> $DIR/rpc_no_ctx.rs:12:32
   |
12 |         pub fn incr(&mut self, ctx: Context, amount: u32) -> Result<()> {
   |                                ^^^

error: aborting due to previous error

